stream-chat-android-offline/src/main/java/io/getstream/chat/android/offline/querychannels/DefaultChatEventHandler.kt

```diff
@@ -6,82 +6,79 @@ import io.getstream.chat.android.client.api.models.QueryChannelsRequest
 import io.getstream.chat.android.client.call.await
 import io.getstream.chat.android.client.events.ChannelUpdatedByUserEvent
 import io.getstream.chat.android.client.events.ChannelUpdatedEvent
-import io.getstream.chat.android.client.events.HasChannel
 import io.getstream.chat.android.client.events.NotificationAddedToChannelEvent
 import io.getstream.chat.android.client.events.NotificationMessageNewEvent
-import io.getstream.chat.android.client.models.Channel
+import io.getstream.chat.android.client.events.NotificationRemovedFromChannelEvent
 import io.getstream.chat.android.client.models.Filters
 import io.getstream.chat.android.client.utils.map
-import kotlinx.coroutines.flow.StateFlow
 import kotlinx.coroutines.runBlocking
 
-internal class DefaultChannelEventsHandler(
-    private val client: ChatClient,
-    private val channels: StateFlow<List<Channel>>,
-) : BaseChannelEventsHandler() {
-    internal var checkFilterOnChannelUpdatedEvent: Boolean = false
+/**
+ * Default implementation of [ChatEventHandler] which is more generic than [MessagingChatEventHandler]. It skips updates
+ * and makes an API request when receives [NotificationAddedToChannelEvent], [NotificationMessageNewEvent],
+ * [NotificationRemovedFromChannelEvent].
+ */
+public open class DefaultChatEventHandler : BaseChatEventHandler() {
 
-    internal var newChannelEventFilter: suspend (Channel, FilterObject) -> Boolean = { channel, filter ->
+    private val client: ChatClient
+        get() = ChatClient.instance()
+
+    internal val newChannelEventFilter: suspend (String, FilterObject) -> Boolean = { cid, filter ->
         client.queryChannels(
             QueryChannelsRequest(
                 filter = Filters.and(
                     filter,
-                    Filters.eq("cid", channel.cid)
+                    Filters.eq("cid", cid)
                 ),
                 offset = 0,
                 limit = 1,
                 messageLimit = 0,
                 memberLimit = 0,
             )
         ).await()
-            .map { channels -> channels.any { it.cid == channel.cid } }
+            .map { channels -> channels.any { it.cid == cid } }
```

Comment:
Again, very hard to understand, what does the `it.data()` mean? can we rename this or somehow give more info to the user what's going on here?

Reply:
It's private and we are not going to expose it to customers. I added some description here

Reply:
Yeah but what if someone else from the team needs to work on it? We need to document all of our code, because otherwise if someone who built the code is not available (e.g. they leave the company), no one knows what's going on anymore

Same for all the KDocs as I mentioned and using more descriptive names...

Reply:
Don't have proper idea how to improve readability here. Just named variable in a closure. Feel free to suggest some improvements here

Reply:
I was more focused on the:

```
 .let { filteringResult -> filteringResult.isSuccess && filteringResult.data() }
```

The `.data()` call is very confusing, but with the KDocs it kind of makes sense I guess. 🤷 I'll resolve the comment, given that it doesn't seem clear to me what we could change to make it more understandable